### PR TITLE
Add `optimize_graph` keyword to `compute`.

### PR DESCRIPTION
**Pull request by jcrist**
If true, the optimizations for each collection won't be run. Default is false. This can be useful for debugging, and mirrors the keyword for `visualize`. Also added docstrings for `compute`.

The keyword was added to the `multiprocessing.get` function as well, as it does `fuse` internally. 
I also removed the unused and untested `optimizations` keyword. This was inconsistent with the other schedulers, didn't seem super necessary, and has been broken for a while (without any complaints, so probably unused). Happy to add it back if others feel it's necessary.

Also fixed up some misuse of `pytest.importorskip` in the `test_base` file, and did some minor pep8 cleanups.

### CODE REVIEW

File: dask/tests/test_base.py [hunk 1]
```diff
@@ -28,81 +37,79 @@ def f3(a):
...
-        assert tokenize(np.array([unicode("Rebeca Alón",encoding="utf-8")], dtype=object)) == \
-               tokenize(np.array([unicode("Rebeca Alón",encoding="utf-8")], dtype=object))
-
+        assert tokenize(np.array([unicode("Rebeca Alón", encoding="utf-8")], dtype=object)) == \
+               tokenize(np.array([unicode("Rebeca Alón", encoding="utf-8")], dtype=object))
 
 
+@pytest.mark.skipif('not np')
 def test_tokenize_numpy_memmap():
-    np = pytest.importorskip('numpy')
```
Comment (mrocklin) [inline dask/tests/test_base.py]: Why these changes?

**Comment by mrocklin** [comment]
Looks fine to me

File: dask/tests/test_base.py (hunk 1 above)
Comment (jcrist) [inline dask/tests/test_base.py]: There were several spots where `importorskip` was used incorrectly. Many tests also relied on several optional packages. When fixing the former I also just cleaned up and standardized all the other instances. Not necessarily better, but more uniform.

**Comment by jcrist** [comment]
Thanks for the review. Merging.
